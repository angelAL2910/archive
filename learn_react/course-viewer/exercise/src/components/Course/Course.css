.create-course{
    margin: auto;
    width: 80%;
}
.create-course select{
    height: 35px;
}
.create-course input, .create-course select{
    width: 100%;
}

.create-course p{
    margin: 2% 0;
}

.create-course button{
color: white;
background-color: #007bff;
margin: 3% 0;
border-radius: 5px;
}

.create-course a{
    color: white;
}