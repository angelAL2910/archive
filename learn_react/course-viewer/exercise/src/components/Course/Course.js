import React from "react";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import "./Course.css";
import { updateCourse } from "../../actions/actionCreators";

export class Course extends React.Component {
  constructor(props) {
    super(props);
    this.titleInput = React.createRef();
    this.categoryInput = React.createRef();
    this.authorInput = React.createRef();
  }

  slugify(url) {
    let wordsInURL = url.split(" ");
    let slugifiedURL = wordsInURL.join("-");
    return slugifiedURL;
  }
  updateCourses(e) {
    e.preventDefault();
    const formData = {};
    for (const field in this.refs) {
      formData[field] = this.refs[field].value;
    }
    let course;
    let { fromEdit, editCourse } = this.props;
    fromEdit ? (course = editCourse[0]) : (course = {});
    course.title = formData["title"];
    course.category = formData["category"];
    course.authorId = formData["author"];
    course.slug = this.slugify(course.title);
    this.props.updateCourse(course);
    this.props.history.push("/courses");
  }
  render() {
    let { authors, fromEdit, editCourse } = this.props;
    let isFromEdit = /true/i.test(fromEdit);
    return (
      <div className="create-course">
        {isFromEdit ? <h2>Edit Course</h2> : <h2>Add Course</h2>}
        <form onSubmit={this.updateCourses.bind(this)}>
          <p>Title</p>
          {isFromEdit ? (
            <input type="text" ref="title" defaultValue={editCourse[0].title} />
          ) : (
            <input type="text" ref="title" />
          )}
          <p>Author</p>
          {isFromEdit ? (
            <select
              id="authorDropdown"
              ref="author"
              defaultValue={editCourse[0].authorId}
            >
              {authors.map((author, id) => (
                <option key={author.id} value={author.id}>
                  {author.name}
                </option>
              ))}
            </select>
          ) : (
            <select id="authorDropdown" ref="author">
              <option key="sel" value="">
                Select an author
              </option>
              {authors.map((author, id) => (
                <option key={author.id} value={author.id}>
                  {author.name}
                </option>
              ))}
            </select>
          )}
          <p>Category</p>
          {isFromEdit ? (
            <input
              type="text"
              ref="category"
              defaultValue={editCourse[0].category}
            />
          ) : (
            <input type="text" ref="category" />
          )}
          <button type="submit">Save</button>
        </form>
      </div>
    );
  }
}

export function mapStateToProps({ courses, authors, editCourse }) {
  return {
    courses,
    authors,
    editCourse
  };
}

export const mapDispatchToPtops = dispatch => ({
  updateCourse: course => {
    dispatch(updateCourse(course));
  }
});

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToPtops
  )(Course)
);
