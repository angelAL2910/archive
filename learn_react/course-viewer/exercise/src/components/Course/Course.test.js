import React from "react";
import renderer from "react-test-renderer";
import { mapStateToProps, mapDispatchToPtops, Course } from "./Course";
import { BrowserRouter } from "react-router-dom";

const fakeStore = {
  courses: [
    {
      id: 1,
      title: "Securing React Apps with Auth01",
      slug: "Securing-React-Apps-with-Auth01",
      authorId: "2",
      category: "JavaScript"
    }
  ],
  authors: [
    {
      id: 1,
      name: "Cory House"
    }
  ],
  editCourse: [
    {
      id: 1,
      title: "Securing React Apps with Auth01",
      slug: "Securing-React-Apps-with-Auth01",
      authorId: "2",
      category: "JavaScript"
    }
  ],
  routing: {
    locationBeforeTransitions: null
  },
};

test("Testing the course page", () => {
  let dispatchedActions = [];

  let newCourse = {
    id: 2,
    title: "Securing React Apps with Auth012",
    slug: "Securing-React-Apps-with-Auth012",
    authorId: "2",
    category: "JavaScript2"
  };
  expect(mapStateToProps(fakeStore).authors.length).toEqual(1);
  const dispatch = jest.fn();
  mapDispatchToPtops(dispatch).updateCourse(newCourse);
  expect(dispatch.mock.calls[0][0]).toEqual({ type: 'UPDATE_COURSE', course: newCourse});
//   expect(dispatchedActions.length).toEqual(1);
  //expect(fakeStore.courses).toContain(newCourse);
});

test("Protecting against regression issues.....", () => {
  let tree = renderer.create(
    <BrowserRouter>
      <Course authors={fakeStore.authors} />
    </BrowserRouter>
  ).toJSON();

  expect(tree).toMatchSnapshot();

});
