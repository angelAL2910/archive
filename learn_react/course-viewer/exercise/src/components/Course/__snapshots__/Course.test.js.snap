// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Protecting against regression issues..... 1`] = `
<div
  className="create-course"
>
  <h2>
    Add Course
  </h2>
  <form
    onSubmit={[Function]}
  >
    <p>
      Title
    </p>
    <input
      type="text"
    />
    <p>
      Author
    </p>
    <select
      id="authorDropdown"
    >
      <option
        value=""
      >
        Select an author
      </option>
      <option
        value={1}
      >
        Cory House
      </option>
    </select>
    <p>
      Category
    </p>
    <input
      type="text"
    />
    <button
      type="submit"
    >
      Save
    </button>
  </form>
</div>
`;
